src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/speaker/SpeakerRouter.java

```diff
@@ -125,34 +127,23 @@ private void speakerMessage(Tuple input, Message message) throws PipelineExcepti
     private void proxySpeaker(Tuple input, Message message) throws PipelineException {
         if (active) {
             if (message instanceof InfoMessage) {
-                proxySpeaker(input, ((InfoMessage) message).getData());
+                proxy(input, ((InfoMessage) message).getData());
             } else {
                 log.error("Do not proxy speaker message - unexpected message type \"{}\"", message.getClass());
             }
         }
     }
 
-    private void proxySpeaker(Tuple input, InfoData payload) throws PipelineException {
-        if (payload instanceof IslInfoData) {
-            emit(STREAM_WATCHER_ID, input, makeWatcherTuple(
-                    input, new WatcherSpeakerDiscoveryCommand((IslInfoData) payload)));
-        } else if (payload instanceof DiscoPacketSendingConfirmation) {
-            emit(STREAM_WATCHER_ID, input, makeWatcherTuple(
-                    input, new WatcherSpeakerSendConfirmationCommand((DiscoPacketSendingConfirmation) payload)));
-        } else if (payload instanceof IslRoundTripLatency) {
-            emit(STREAM_WATCHER_ID, input, makeWatcherTuple(
-                    input, new WatcherSpeakerRoundTripDiscovery((IslRoundTripLatency) payload)));
-        } else if (payload instanceof SwitchInfoData) {
-            emit(input, makeDefaultTuple(input, new SwitchEventCommand((SwitchInfoData) payload)));
-        } else if (payload instanceof PortInfoData) {
-            emit(input, makeDefaultTuple(input, new SwitchPortEventCommand((PortInfoData) payload)));
-        } else if (payload instanceof NetworkDumpSwitchData) {
-            emit(input, makeDefaultTuple(
-                    input, new SwitchManagedEventCommand(((NetworkDumpSwitchData) payload).getSwitchView())));
-        } else if (payload instanceof UnmanagedSwitchNotification) {
-            emit(input, makeDefaultTuple(
-                    input, new SwitchUnmanagedEventCommand(((UnmanagedSwitchNotification) payload).getSwitchId())));
-        } else if (payload instanceof DeactivateSwitchInfoData) {
+    private void proxy(Tuple input, InfoData payload) throws PipelineException {
+        if (! proxyUnconditionally(input, payload)) {
+            if (! proxyOnlyWhenActive(input, payload)) {
```

Comment:
extra spaces?

Reply:
em... have I mixed up java's coding style with python's?.. I was sure we must have a space after `!` operator.